### PR TITLE
[KernelGen][Nvidia] Add special_chebyshev_polynomial_t operator with Triton kernel

### PR DESCRIPTION
**Pull request by CoyeCHEN**
## Summary

Adds the `special_chebyshev_polynomial_t` operator with its implementation, accuracy tests and benchmarks. Every registered key is verified to map to a live ATen overload.

Target registrations added or remapped: `default`, `n_scalar`, `out`.

## Overload Coverage

| ATen overload | status | FlagGems wrapper | evidence |
|---|---|---|---|
| `default` | `add` | `special_chebyshev_polynomial_t` | GEMS SPECIAL_CHEBYSHEV_POLYNOMIAL_T |
| `n_scalar` | `add` | `special_chebyshev_polynomial_t` | GEMS SPECIAL_CHEBYSHEV_POLYNOMIAL_T |
| `out` | `add` | `special_chebyshev_polynomial_t_out` | GEMS SPECIAL_CHEBYSHEV_POLYNOMIAL_T_OUT |

## Testing

- Direct ATen dispatch tests: 15 passed, 0 failed, 0 skipped.
- Complete changed test file on GPU: 16 passed, 0 skipped, 0 failed.
- Target ATen tests in CPU quick-reference mode: 3 passed, 0 failed, 0 skipped.
- Complete changed test file in CPU quick-reference mode: 4 passed, 0 skipped, 0 failed.
- Complete core benchmark file: 3 passed, 0 skipped, 0 failed.
- Base: `8bee1062b6fdb514f3544882f2463b05ee8bb5d2` (`flagos-ai/FlagGems:master`).

### `special_chebyshev_polynomial_t`

| dtype | shape | Torch (ms) | FlagGems (ms) | speedup | TFLOPS |
|---|---:|---:|---:|---:|---:|
| float32 | `1024, 1024` | 0.019072 | 0.070976 | 0.269 | 0.000 |
| float32 | `20, 320, 15` | 0.010656 | 0.043392 | 0.246 | 0.000 |
| float32 | `16, 128, 64, 1280` | 1.702048 | 10.871232 | 0.157 | 0.000 |
| float32 | `2, 19, 7` | 0.007392 | 0.041728 | 0.177 | 0.000 |

Complete result count: 4; geometric mean speedup: **0.207x**.

### `special_chebyshev_polynomial_t_n_scalar`

| dtype | shape | Torch (ms) | FlagGems (ms) | speedup | TFLOPS |
|---|---:|---:|---:|---:|---:|
| float32 | `1024, 1024` | 0.010080 | 0.172832 | 0.058 | 0.000 |
| float32 | `20, 320, 15` | 0.007584 | 0.135168 | 0.056 | 0.000 |
| float32 | `16, 128, 64, 1280` | 0.581520 | 11.021712 | 0.053 | 0.000 |
| float32 | `2, 19, 7` | 0.006144 | 0.131840 | 0.047 | 0.000 |

Complete result count: 4; geometric mean speedup: **0.053x**.

### `special_chebyshev_polynomial_t_out`

| dtype | shape | Torch (ms) | FlagGems (ms) | speedup | TFLOPS |
|---|---:|---:|---:|---:|---:|
| float32 | `1024, 1024` | 0.019040 | 0.064320 | 0.296 | 0.000 |
| float32 | `20, 320, 15` | 0.010592 | 0.028320 | 0.374 | 0.000 |
| float32 | `16, 128, 64, 1280` | 1.701776 | 10.872928 | 0.157 | 0.000 |
| float32 | `2, 19, 7` | 0.007392 | 0.027776 | 0.266 | 0.000 |

Complete result count: 4; geometric mean speedup: **0.261x**.

## Multi-backend Testing

| Backend | Accuracy | Benchmark | Notes |
|---|---|---|---|
| Nvidia H20 | PASS (15 direct / 16 full-file GPU / 4 quick-CPU tests) | PASS (12 complete target rows) | Primary validation |
| Other backends | N/A | N/A | Not run in this local pipeline |

## Files Changed

- `benchmark/test_special_chebyshev_polynomial_t.py`: +106 / -0
- `conf/operators.yaml`: +28 / -0
- `src/flag_gems/__init__.py`: +3 / -0
- `src/flag_gems/ops/__init__.py`: +6 / -0
- `src/flag_gems/ops/special_chebyshev_polynomial_t.py`: +87 / -0
- `tests/test_special_chebyshev_polynomial_t.py`: +108 / -0

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -9436,6 +9436,21 @@ ops:
       - Math
     stages:
       - alpha: '5.4'
+  - id: special_chebyshev_polynomial_t
```
Comment (Caeruleann) [inline conf/operators.yaml]:
```suggestion
  - id: special_chebyshev_polynomial_t
  - id: special_chebyshev_polynomial_t_out
```

add id for `special_chebyshev_polynomial_t_out`

Comment (CoyeCHEN) [reply conf/operators.yaml]: Good catch! Checking the precedent from `special_chebyshev_polynomial_w` (#5352, merged), the `.out` variant should indeed have a **separate id entry** rather than being listed in the same `for` array.

Current master shows:
```yaml
- id: special_chebyshev_polynomial_w       # base operator
  for: [special.chebyshev_polynomial_w]
- id: special_chebyshev_polynomial_w_out   # separate entry for .out
  for: [special.chebyshev_polynomial_w.out]
```

I'll update this PR to follow the same pattern:
```yaml
- id: special_chebyshev_polynomial_t
  for: 
    - special_chebyshev_polynomial_t
    - special_chebyshev_polynomial_t.n_scalar
- id: special_chebyshev_polynomial_t_out
  for: [special_chebyshev_polynomial_t.out]
```

This requires adding the corresponding `_out` wrapper, test cases, and benchmark. Will push the update shortly.

File: src/flag_gems/ops/special_chebyshev_polynomial_t.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    ATen wrapper: special_chebyshev_polynomial_t.out(Tensor x, Scalar n, *, Tensor(a!) out) -> Tensor(a!)
+    """
+    logger.debug("GEMS SPECIAL_CHEBYSHEV_POLYNOMIAL_T_OUT")
+    if x.dtype not in (torch.float32, torch.float64):
+        raise ValueError(
+            f"special_chebyshev_polynomial_t only supports float32/float64, got {x.dtype}"
+        )
+    if not isinstance(n, torch.Tensor):
+        n = torch.tensor(n, dtype=torch.int64, device=x.device)
+    return chebyshev_polynomial_t_func(x, n, out0=out)
```
Comment (Caeruleann) [inline src/flag_gems/ops/special_chebyshev_polynomial_t.py:87]: who used `out`

Comment (CoyeCHEN) [reply src/flag_gems/ops/special_chebyshev_polynomial_t.py:87]: The `.out` variant is used by PyTorch when an output tensor is pre-allocated. The operator now registers both `special_chebyshev_polynomial_t.default` and `special_chebyshev_polynomial_t_out` (as shown in operators.yaml and the ci_test_aliases.yaml mapping).

The implementation uses `n_scalar_out` wrapper which forwards to the base function, following the standard pattern for _out variants in this codebase. You can verify it dispatches correctly by checking that both registration keys appear in the dispatch log.

**Review by Caeruleann — APPROVED** [review]
